arkouda/segarray.py

```diff
@@ -803,5 +803,213 @@ def load(cls, prefix_path, dataset='segarray', segment_suffix='_segments', value
+        SegArray([
+        [1, 2, 3, 4, 5],
+        [1, 2, 3, 4, 5]
+        ])
+        """
+        from arkouda.pdarraysetops import union1d
+
+        a_seg_inds = self.grouping.broadcast(arange(self.size)[self._non_empty])
+        b_seg_inds = other.grouping.broadcast(arange(other.size)[other._non_empty])
+        (new_seg_inds, new_values) = union1d([a_seg_inds, self.values], [b_seg_inds, other.values])
+        g = GroupBy(new_seg_inds)
+        # This method does not return any empty resulting segments. We need to add these if they are missing.
+        if g.segments.size == self.size:
+            return SegArray(g.segments, new_values[g.permutation])
+        else:
+            segments = zeros(self.size, dtype=akint64)
+            k, ct = g.count()
+            segments[k] = g.segments
+            truth = ~self._non_empty | ~other._non_empty
+            if truth[-1]:
+                segments[-1] = g.permutation.size
+                truth[-1] = False
+            segments[truth] = segments[arange(self.size)[truth] + 1]
+            return SegArray(segments, new_values[g.permutation])
+
+    def setdiff_ma(self, other):
+        """
+        Computes the set difference of 2 SegArrays.
+
+        Parameters
+        ----------
+        other : SegArray
+            SegArray to compute against
+
+        Returns
+        -------
+        SegArray
+            Segments are the 1d set difference of the segments of self and other
+
+        See Also
+        --------
+        pdarraysetops.setdiff1d
+
+        Examples
+        --------
+        >>> a = [1, 2, 3, 1, 4]
+        >>> b = [3, 1, 4, 5]
+        >>> c = [1, 3, 3, 5]
+        >>> d = [2, 2, 4]
+        >>> seg_a = ak.SegArray(ak.array([0, len(a)]), ak.array(a+b))
+        >>> seg_b = ak.SegArray(ak.array([0, len(c)]), ak.array(c+d))
+        >>> seg_a.setdiff_ma(seg_b)
+        SegArray([
+        [2, 4],
+        [1, 3, 5]
+        ])
+        """
+        from arkouda.pdarraysetops import setdiff1d
+
+        a_seg_inds = self.grouping.broadcast(arange(self.size)[self._non_empty])
+        b_seg_inds = other.grouping.broadcast(arange(other.size)[other._non_empty])
+        (new_seg_inds, new_values) = setdiff1d([a_seg_inds, self.values], [b_seg_inds, other.values])
+        g = GroupBy(new_seg_inds)
+        # This method does not return any empty resulting segments. We need to add these if they are missing.
+        if g.segments.size == self.size:
+            return SegArray(g.segments, new_values[g.permutation])
+        else:
+            segments = zeros(self.size, dtype=akint64)
+            k, ct = g.count()
+            segments[k] = g.segments
+            truth = ~self._non_empty | ~other._non_empty
+            if truth[-1]:
+                segments[-1] = g.permutation.size
+                truth[-1] = False
+            segments[truth] = segments[arange(self.size)[truth] + 1]
+            return SegArray(segments, new_values[g.permutation])
```

Comment:
found a small case for `setdiff` with multiple empty segments that isn't correctly handled. I let ethan know and he has a fix